### PR TITLE
Feature/pf 130

### PR DESCRIPTION
**Pull request by Molvilada**
Validaciones formulario login

## Release Notes

- **New Features**
	- Introduced a `LoginViewModel` to manage login state and validation.
	- Added a `LoginScreen` that integrates with the `LoginViewModel` for improved user input handling.
	- Enhanced `SignUpScreen` with better string resource management.

- **Bug Fixes**
	- Improved validation logic in both `LoginViewModel` and `SignUpViewModel`.

- **Tests**
	- Added unit tests for `LoginViewModel` to ensure proper validation.
	- Refactored tests for `SignUpViewModel` to simplify error message handling.

- **Chores**
	- Updated dependency resolution management for improved plugin handling.

### CODE REVIEW

File: app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt [hunk 1]
```diff
@@ -130,15 +117,15 @@ fun LoginScreen(navController: NavController) {
 
                 val annotatedString = buildAnnotatedString {
                     withStyle(style = SpanStyle(fontWeight = FontWeight.Light)) {
-                        append(context.getString(R.string.login_signup_question) + " ")
+                        append(stringResource(R.string.login_signup_question) + " ")
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt:120]: Si le pongo el espacio no lo respeta

File: app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt [hunk 2]
```diff
@@ -60,54 +59,42 @@ fun LoginScreen(navController: NavController) {
...
-                    val validationResults = validateFields(
-                        context = context,
-                        email = email,
-                        password = password,
-                    )
-
-                    emailError = validationResults["emailError"] ?: ""
-                    passwordError = validationResults["passwordError"] ?: ""
+                    val isValid =
+                        viewModel.validateFields(context.getString(R.string.form_required))
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt:95]: No, si lo hago asi me sale este error: @Composable invocations can only happen from the context of a @Composable functio

File: app/src/main/java/io/capibaras/abcall/ui/views/SignUpScreen.kt [hunk 3]
```diff
@@ -138,7 +139,12 @@ fun SignUpScreen(navController: NavController) {
 
             Button(
                 onClick = {
-                    val isValid = viewModel.validateFields(context)
+                    val isValid = viewModel.validateFields(
+                        context.getString(R.string.form_required),
+                        context.getString(R.string.form_invalid_email),
+                        context.getString(R.string.form_password_length),
+                        context.getString(R.string.form_confirm_password_invalid)
+                    )
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/ui/views/SignUpScreen.kt:147]: No, si lo hago asi me sale este error: `@Composable` invocations can only happen from the context of a `@Composable` function

File: app/src/test/java/io/capibaras/abcall/LoginViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        viewModel.email = "johndoe@gmail.com"
+        viewModel.password = "password123"
+
+        val isValid = viewModel.validateFields(requiredMsg)
+
+        assertTrue(isValid)
+        assertEquals("", viewModel.emailError)
+        assertEquals("", viewModel.passwordError)
+    }
+}
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/test/java/io/capibaras/abcall/LoginViewModelTest.kt:42]: No existe esa validación en la pantalla de login

File: app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt [hunk 5]
```diff
@@ -39,10 +35,13 @@ import io.capibaras.abcall.R
 import io.capibaras.abcall.ui.components.CustomOutlinedTextField
 import io.capibaras.abcall.ui.theme.ABCallTheme
 import io.capibaras.abcall.ui.theme.linkText
+import io.capibaras.abcall.viewmodels.LoginViewModel
+import org.koin.androidx.compose.koinViewModel
 
 @Composable
 fun LoginScreen(navController: NavController) {
     val context = LocalContext.current
+    val viewModel: LoginViewModel = koinViewModel()
```
[the remark this answers is not in the document]
Comment (Molvilada) [reply app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt:44]: La idea de usar koin es no tener que pasa el viewmodel como parametro

File: app/src/main/java/io/capibaras/abcall/ui/views/SignUpScreen.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (tpambor) [reply app/src/main/java/io/capibaras/abcall/ui/views/SignUpScreen.kt:147]: @coderabbitai this is in the onClick handler which is not a composable function (at the moment) and produces the following error: `@Composable invocations can only happen from the context of a @Composable function`

File: app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (tpambor) [reply app/src/main/java/io/capibaras/abcall/ui/views/LoginScreen.kt:44]: @Molvilada Más común es hacerlo así
`fun LoginScreen(navController: NavController, viewModel: LoginViewModel = koinViewModel())` pero al final casi lo mismo, puedes hacer como tu quieres.

**Review by tpambor — APPROVED** [review]

**Pull request merged**